cypress-io/cypress
# fix: Subscribe to framework detection changes in wizard

- Closes #26388 

### Additional details
Launchpad does not wait for the data context to initialize in case there are significant slowdowns which would make the UI non-responsive on launch. This can cause the CT framework auto-detection to not populate if the detection is somehow delayed (complex project, slow machine, etc) - the detection occurs correctly but the GQL query in the Vue component has already resolved so the result is not reflected in the UI.

This PR adds a subscription so the Vue component can listen for updates to this data and reflect them in the UI.

### Steps to test
1. On `develop`, add a delay to wizard initialization by updating the `initialize` function with this line:
```
// WizardActions.ts
async initialize () {
  await new Promise((resolve) => setTimeout(resolve, 5000))
  ...
```
2. Create a new empty CT project: `npm create vite@latest`, use any supported framework & options
3. Install project dependencies and add Cypress: `cd <project dir>`, `npm i`, `npm i -D cypress`
3. Open project directly into CT mode: `cd <cypress repo> `, `yarn cypress:open --project <path to project> --component`
4. Observe that UI does not detect framework/bundler, stays on "Pick a framework"
5. Switch to this branch, repeat procedure
6. Observe framework & bundler are reflected in the UI after small delay

### How has the user experience changed?
**Before**

https://user-images.githubusercontent.com/11482842/230137927-c24db131-3603-4274-a3b7-d4fa7092d427.mov

**After**

https://user-images.githubusercontent.com/11482842/230137976-0895f4a2-2994-40f9-a8d1-2aea6081ca6d.mov

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
